### PR TITLE
[video_player_videohole] Merge media player code from avplay

### PR DESCRIPTION
**Pull request by xiaowei-guan**
There are still third-party using the video_player_videohole plugin because they are not partner. Compared to avplay, video_player_videohole's code is older, so merege media player code from avplay.

### CODE REVIEW

**Review by JSUYA — COMMENTED** [review]
I left a few comments.

+)
in video_player_tizen.dart, wrong comment
```
- /// Pigeon-generated [TizenVideoPlayerApi].
+ /// Pigeon-generated [VideoPlayerVideoholeApi].
```

+)
In packages/video_player_videohole/pubspec.yaml.
`ffi` package dependency does not seem to be needed any more. Do we need it?

File: packages/video_player_videohole/tizen/src/drm_manager.cc [hunk 1]
```diff
@@ -4,248 +4,326 @@
...
-    return false;
-  }
-  return true;
+  return DMGRSecurityInitCompleteCB(drm_handle, len, pssh_data,
+                                    &security_param);
 }
 
-bool DrmManager::SetChallengeCondition() {
+int DrmManager::SetChallenge(const std::string &media_url) {
+  LOG_ERROR("[DrmManager] SetChallenge url = %s", media_url.c_str());
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply packages/video_player_videohole/tizen/src/drm_manager.cc]: Fixed

File: packages/video_player_videohole/tizen/src/video_player_tizen_plugin.cc [hunk 2]
```diff
@@ -147,18 +144,36 @@ ErrorOr<PlayerMessage> VideoPlayerTizenPlugin::Create(
...
   }
 
-  int64_t player_id =
-      player->Create(uri, drm_type, license_server_url, http_headers);
+  int64_t player_id = 0;
+  auto player = std::make_unique<MediaPlayer>(
+      plugin_registrar_->messenger(),
+      FlutterDesktopPluginRegistrarGetView(registrar_ref_));
+  player_id = player->Create(uri, drm_type, license_server_url, prebuffer_mode,
+                             http_headers);
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply packages/video_player_videohole/tizen/src/video_player_tizen_plugin.cc]: Fixed

**Comment by xiaowei-guan** [comment]
> I left a few comments.
> 
> +) in video_player_tizen.dart, wrong comment
> 
> ```
> - /// Pigeon-generated [TizenVideoPlayerApi].
> + /// Pigeon-generated [VideoPlayerVideoholeApi].
> ```
> 
> +) In packages/video_player_videohole/pubspec.yaml. `ffi` package dependency does not seem to be needed any more. Do we need it?

I have removed ffi package and modify video_player_tizen.dart

**Review by JSUYA — APPROVED** [review]

**Pull request merged**